backend/src/database/repositories/memberAttributeSettingsRepository.ts

```diff
@@ -0,0 +1,230 @@
+import lodash from 'lodash'
+import Sequelize from 'sequelize'
+import SequelizeRepository from './sequelizeRepository'
+import { IRepositoryOptions } from './IRepositoryOptions'
+import Error404 from '../../errors/Error404'
+import SequelizeFilterUtils from '../utils/sequelizeFilterUtils'
+
+const Op = Sequelize.Op
+
+class MemberAttributeSettingsRepository {
+  static async findById(id, options: IRepositoryOptions) {
+    const transaction = SequelizeRepository.getTransaction(options)
+
+    const include = []
+
+    const currentTenant = SequelizeRepository.getCurrentTenant(options)
+
+    const record = await options.database.memberAttributeSettings.findOne({
+      where: {
+        id,
+        tenantId: currentTenant.id,
+      },
+      include,
+      transaction,
+    })
+
+    if (!record) {
+      throw new Error404()
+    }
+
+    return this._populateRelations(record)
+  }
+
+  static async _populateRelations(record) {
+    if (!record) {
+      return record
+    }
+    const output = record.get({ plain: true })
+
+    return output
+  }
+
+  static async create(data, options: IRepositoryOptions) {
+    const currentUser = SequelizeRepository.getCurrentUser(options)
+
+    const tenant = SequelizeRepository.getCurrentTenant(options)
+
+    const transaction = SequelizeRepository.getTransaction(options)
+
+    const record = await options.database.memberAttributeSettings.create(
+      {
+        ...lodash.pick(data, ['type', 'name', 'label', 'canDelete', 'show']),
```

Comment:
This is not very typescripty :) I would suggest introducing an interface for `data` parameter and instead of doing `...lodash.pick` I would just set the property one by one - that way if we change the interface in the future the build will fail here if the property of the interface is removed for instance.

[Here](https://github.com/CrowdDotDev/crowd.dev/pull/43/files#diff-965f8e04f52743e99d3dcf965807f68f2d41ae8d7e726830eb77a69a4d45a524R22) is my implementation of `automationRepository` for reference.